Upgrades Angular static libraries

NishealJ opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
This PR Upgrades :
 **Message Format** 
**angularTranslate**, 
**angularCookiesRev**, 
**angularStorageCookiesRev**,
 **angularTranslateInterpolationMessageFormat**, 
**angularTranslateLoader**
and **bowerAngularTranslateLoaderPartial** to their latest version 
please visit [here](https://docs.google.com/document/d/1MD3vSZEEJ_zwyjqbdd2b5BmWwzn098wjaVfJQ4dLQa8/edit?usp=sharing) for the testing doc 
## Checklist
- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The linter/Karma presubmit checks have passed.
  - These should run automatically, but if not, you can manually trigger them locally using `python scripts/pre_commit_linter.py` and `bash scripts/run_frontend_tests.sh`.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR has an appropriate "CHANGELOG: ..." label (If you are unsure of which label to add, ask the reviewers for guidance).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide).
- [x] The PR addresses the points mentioned in the codeowner checks for the files/folders changed. (See the [codeowner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).)
- [x] The PR is **assigned** to an appropriate reviewer.
  - If you're a new contributor, please ask on [Gitter](https://gitter.im/oppia/oppia-chat) for someone to assign a reviewer and don't tick this checkbox.
  - If you're not sure who the appropriate reviewer is, please assign to the issue's "owner" -- see the "talk-to" label on the issue. Do not only request the review but also add the reviewer as an assignee.

NishealJ committed:
initial upgrade

NishealJ committed:
fix

apb7 commented:
Hi @NishealJ, can you please provide the link to the testing doc? Thanks!

NishealJ commented:
> Hi @NishealJ, can you please provide the link to the testing doc? Thanks!

Hi @apb7, I've added it but needs some suggestion for areas to brief more while testing Thanks! 

Added doc testing [here](https://docs.google.com/document/d/1MD3vSZEEJ_zwyjqbdd2b5BmWwzn098wjaVfJQ4dLQa8/edit)

apb7 commented:
Hi @NishealJ, the testing doc looks a bit general. Can you please look into the changelog and then analyze the breaking changes? Also, adding @DubeySandeep for suggestions regarding testing Angular translate and other related libraries. Thanks!

NishealJ commented:
> Hi @NishealJ, the testing doc looks a bit general. Can you please look into the changelog and then analyze the breaking changes? Also, adding @DubeySandeep for suggestions regarding testing Angular translate and other related libraries. Thanks!

Hi @apb7 & @DubeySandeep, I've added more tests but still I've less ideas to test the static libraries
need @DubeySandeep's suggestions :) Thanks!

DubeySandeep commented:
@NishealJ, I've left a few comments on the doc for testing angularTranslate and $cookies behavior PTAL! (btw, angular translate isn't used in the "Translation and voiceover" project. :p )

DubeySandeep commented on manifest.json:
```diff
@@ -167,29 +167,29 @@
+        "version": "2.18.1",
         "downloadFormat": "zip",
-        "url": "https://github.com/angular-translate/bower-angular-translate/archive/2.8.1.zip",
+        "url": "https://github.com/angular-translate/bower-angular-translate/archive/2.18.1.zip",
```
@apb7, Can we use[ cdnjs](https://cdnjs.com/libraries/angular-translate) and only download the files which are required?  (not sure!)
Asking this:
- As downloading files through CDNs use to be fast and we can download specific files which will be less in size.

NishealJ replied:
yes we can have a try probably with lightweight static libraries

NishealJ replied:
Hi @DubeySandeep, I've added the cdnjs for the angular static libs PTAL Thanks!

NishealJ commented:
> @NishealJ, I've left a few comments on the doc for testing angularTranslate and $cookies behavior PTAL! (btw, angular translate isn't used in the "Translation and voiceover" project. :p )

That's why I was tagging you for a long time here because I knew I would end-up testing all the translations modules of oppia for a static lib, And the test-cases which you will suggest will be on-point :D

NishealJ committed:
Add cdn for static libs

seanlip commented:
Code looks fine to me, I left a couple of comments on the testing doc.

apb7 reviewed (approved):
LGTM code wise but please resolve all comments on the testing doc. Thanks, @NishealJ!

vojtechjelinek reviewed (approved):
LGTM from code owner's perspective. Thanks!

NishealJ committed:
sync angular cookie -v

NishealJ commented:
> Code looks fine to me, I left a couple of comments on the testing doc.

Hi @seanlip & @apb7, I've addressed all your comments in the doc and changed the angular-cookie version to match angular version PTAL Thanks

seanlip reviewed (approved):
PR looks good. Please resolve remaining comments in the testing doc (I think you missed one) and don't commit package-lock.json. Thanks!

NishealJ committed:
Revert package-lock

NishealJ committed:
Update with develop

NishealJ committed:
Merge remote-tracking branch 'origin/develop' into ang_static_up
